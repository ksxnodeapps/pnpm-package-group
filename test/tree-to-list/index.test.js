'use strict'
const path = require('path')
const fs = require('fs')
const yaml = require('js-yaml')
const convert = require('../../lib/tree-to-list')

describe('exported entity', () => {
  it('is a function', () => expect(typeof convert).toBe('function'))

  describe('when being called with input.yaml', () => {
    const readYamlFile = basename =>
      yaml.safeLoad(fs.readFileSync(path.resolve(__dirname, basename), 'utf8'))

    const input = readYamlFile('input.yaml')
    const output = readYamlFile('output.yaml')

    it('gives result equal to output.yaml', () => {
      expect(convert(input)).toEqual(output)
    })

    it('should stay unchanged', () => {
      expect(convert(input)).toMatchSnapshot()
    })
  })
})
