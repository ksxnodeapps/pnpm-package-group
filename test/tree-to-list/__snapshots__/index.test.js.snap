// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`exported entity when being called stays unchanged 1`] = `
Object {
  "flat": Object {
    "description": "Flat (explicitly set __flat to true)",
    "input": Object {
      "__flat": true,
      "abc": Array [
        "lorem",
        "ipsum",
      ],
      "def": Object {
        "bar": Array [
          "consectetur",
        ],
        "foo": Array [
          "dolor",
          "sit",
          "amet",
        ],
      },
    },
    "output": Array [
      Object {
        "data": Array [
          "lorem",
          "ipsum",
          "dolor",
          "sit",
          "amet",
          "consectetur",
        ],
        "path": "",
      },
    ],
  },
  "mix": Object {
    "description": "Mixing nested and flat together",
    "input": Object {
      "__flat": false,
      "abc": Array [
        "lorem",
        "ipsum",
      ],
      "def": Object {
        "__flat": true,
        "bar": Array [
          "consectetur",
        ],
        "baz": Object {
          "global": Object {
            "__flat": true,
            "abc": Array [
              "loremzzz",
              "ipsumzzz",
            ],
            "def": Object {
              "bar": Array [
                "consecteturzzz",
              ],
              "foo": Array [
                "dolorzzz",
                "sitzzz",
                "ametzzz",
              ],
            },
          },
          "inherit": Object {
            "abc": Array [
              "loremz",
              "ipsumz",
            ],
            "def": Object {
              "bar": Array [
                "consecteturz",
              ],
              "foo": Array [
                "dolorz",
                "sitz",
                "ametz",
              ],
            },
          },
          "local": Object {
            "__flat": false,
            "abc": Array [
              "loremzz",
              "ipsumzz",
            ],
            "def": Object {
              "bar": Array [
                "consecteturzz",
              ],
              "foo": Array [
                "dolorzz",
                "sitzz",
                "ametzz",
              ],
            },
          },
        },
        "foo": Array [
          "dolor",
          "sit",
          "amet",
        ],
      },
    },
    "output": Array [
      Object {
        "data": Array [
          "lorem",
          "ipsum",
        ],
        "path": "abc/",
      },
      Object {
        "data": Array [
          "dolor",
          "sit",
          "amet",
          "consectetur",
          "loremz",
          "ipsumz",
          "dolorz",
          "sitz",
          "ametz",
          "consecteturz",
          "loremzzz",
          "ipsumzzz",
          "dolorzzz",
          "sitzzz",
          "ametzzz",
          "consecteturzzz",
        ],
        "path": "def/",
      },
      Object {
        "data": Array [
          "loremzz",
          "ipsumzz",
        ],
        "path": "def/baz/local/abc/",
      },
      Object {
        "data": Array [
          "dolorzz",
          "sitzz",
          "ametzz",
        ],
        "path": "def/baz/local/def/foo/",
      },
      Object {
        "data": Array [
          "consecteturzz",
        ],
        "path": "def/baz/local/def/bar/",
      },
    ],
  },
  "nested": Object {
    "description": "Nested (explicitly set __flat to false)",
    "input": Object {
      "__flat": false,
      "abc": Array [
        "lorem",
        "ipsum",
      ],
      "def": Object {
        "bar": Array [
          "consectetur",
        ],
        "foo": Array [
          "dolor",
          "sit",
          "amet",
        ],
      },
    },
    "output": Array [
      Object {
        "data": Array [
          "lorem",
          "ipsum",
        ],
        "path": "abc/",
      },
      Object {
        "data": Array [
          "dolor",
          "sit",
          "amet",
        ],
        "path": "def/foo/",
      },
      Object {
        "data": Array [
          "consectetur",
        ],
        "path": "def/bar/",
      },
    ],
  },
  "noisy": Object {
    "description": "Ignore names begin with '_'",
    "input": Object {
      "__noise": 123,
      "_noise": 456,
      "abc": Array [
        "lorem",
        "ipsum",
      ],
      "def": Object {
        "bar": Array [
          "consectetur",
        ],
        "foo": Array [
          "dolor",
          "sit",
          "amet",
        ],
      },
    },
    "output": Array [
      Object {
        "data": Array [
          "lorem",
          "ipsum",
        ],
        "path": "abc/",
      },
      Object {
        "data": Array [
          "dolor",
          "sit",
          "amet",
        ],
        "path": "def/foo/",
      },
      Object {
        "data": Array [
          "consectetur",
        ],
        "path": "def/bar/",
      },
    ],
  },
  "simple": Object {
    "description": "Simplest form (nested)",
    "input": Object {
      "abc": Array [
        "lorem",
        "ipsum",
      ],
      "def": Object {
        "bar": Array [
          "consectetur",
        ],
        "foo": Array [
          "dolor",
          "sit",
          "amet",
        ],
      },
    },
    "output": Array [
      Object {
        "data": Array [
          "lorem",
          "ipsum",
        ],
        "path": "abc/",
      },
      Object {
        "data": Array [
          "dolor",
          "sit",
          "amet",
        ],
        "path": "def/foo/",
      },
      Object {
        "data": Array [
          "consectetur",
        ],
        "path": "def/bar/",
      },
    ],
  },
}
`;
